### PR TITLE
GitHub basics

### PR DESCRIPTION
**Pull request by VerisimilitudeX**
@ozhang22 https://github.com/VerisimilitudeX/NF-SampleRepo

### CODE REVIEW

**Comment by VerisimilitudeX** [comment]
@ozhang22 thoughts on the README so far? https://github.com/VerisimilitudeX/NF-SampleRepo

We can move it to the Nuevo Foundation org once I finish.

File: content/english/github-basics/activity-3.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Congratulations, brave coders! You've successfully set up your GitHub repository – the headquarters for your coding expedition with Maria and Kyle. 🎉🏰
+
+Now, it's time to stock your repository with precious files and enchanted code! Let's learn the two ways to add files to it! ✨📂
+
+First, go into your new repository, click on the "Add file" button, and select "Create new file".
+- Now you can name your new file and start coding! Below is a photo of a file created by Maria to say hello to Kyle:
+
+<img alt="Creating a new file in the GitHub repository named hello_world.py and adding code to it" src="../images/NewFile.png" height="400"/>
+
+- After you've finish coding, scroll down to the bottom of the page and you will see this:
```
Comment (ozhang22) [inline content/english/github-basics/activity-3.md:17]: Can we have some sample code that we will add to Maria and Kyle's project?

File: content/english/github-basics/activity-3.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+<img alt="Commit options for the new file" src="../images/commit_new.PNG" height="400"/>
+
+<img alt="GitHub's repository view with hello_world.py showing in the list of files" src="../images/tempsnip4.png" height="400"/>
+
+<img alt="File upload options in the GitHub repository" src="../images/UploadNewFile.png" height="400"/>
+
+# Your GitHub Repository Blooms! 🌱🏞️
```
Comment (ozhang22) [inline content/english/github-basics/activity-3.md:24]: Due to accessibility, we can't use # (h1) for any of the headers, all headers must be at least ## (h2)

```suggestion
## Your GitHub Repository Blooms! 🌱🏞️
```

File: content/english/github-basics/activity-3.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+With your newfound abilities, your GitHub repository will flourish, each file a magical blossom in your coding garden! 🌷🌼
+
+As you and your companions continue your coding quest, remember to add and commit your changes often. This way, you'll be able to go back in time and view your code in the past.
```
Comment (ozhang22) [inline content/english/github-basics/activity-3.md:28]: Do we have a section on how to look at the commit history?

File: content/english/github-basics/activity-5.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+---
+title: "Navigating with GitHub Issues"
+date: 2023-08-10T11:00:00-07:00
+draft: false
+weight: 6
+---
+
+## Navigating with GitHub Issues: Building Your Coding Boat
+
+Imagine you and your team are setting sail on a coding adventure across a vast digital ocean. Your goal is to build a sturdy and innovative coding boat that can weather challenges and reach new horizons. But every voyage encounters storms and rough waters. This is where GitHub Issues come in – they're your tools for crafting and repairing your coding boat as you navigate the digital waves.
```
Comment (ozhang22) [inline content/english/github-basics/activity-5.md:10]: Can we incorporate Maria and Kyle into this?

File: content/english/github-basics/activity-6.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+### The Power of Continuous Integration (CI)
+
+Think of Continuous Integration as your boat's quality control mechanism. Every time you or your team members push code to the repository, CI automatically tests and integrates changes. This prevents potential issues from growing into full-blown storms.
+
+### Setting Up GitHub Actions and CI
+
+Getting GitHub Actions and CI on board is straightforward:
+
+1. **Creating a Workflow**:
```
Comment (ozhang22) [inline content/english/github-basics/activity-6.md:24]: Can we set up an example action to use for Maria and Kyle's repo?

File: content/english/github-basics/activity-5.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+### Crafting Your Coding Boat
+
+Creating an issue is like drawing up blueprints for a boat component:
+
+1. **From GitHub Web Interface**:
+   - Open your repository on GitHub.
+   - Navigate to the "Issues" tab.
+   - Click "New Issue."
+   - Craft a clear title and provide details about the boat component (issue).
```
Comment (ozhang22) [inline content/english/github-basics/activity-5.md:24]: Can we give an example of an issue found in Maria and Kyle's workshop?

**Review by ozhang22 — COMMENTED** [review]
Added some initial comments. As discussed previously I would like to see more integration with Kyle and Maria's story, and also have some concrete examples of the tutorials so that students understand the benefits of these features

File: content/english/github-basics/activity-1.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+2. When you get to the website, it should look like the photo below:
+<img alt="GitHub.com homepage with the sign up box highlighted" src="../images/CreatingRepositorySignUp.png" height="400"/>
+
+    - On the top right corner of the website, you should see two buttons: "Sign In" and "Sign Up". If you do not have an account, click on the "Sign Up" button and follow the directions given, for example:
+
+<img alt="GitHub sign up page with email and password entered" src="../images/SignupPage.jpg
+" height="400"/>
+
+3. Congratulations! You now have a GitHub account! Now, let's make sure you can log in...
```
[the remark this answers is not in the document]
Comment (VerisimilitudeX) [reply content/english/github-basics/activity-1.md:26]: I believe that it doesn't send an activation email, but I added "email" to the prerequisites.

**Comment by ozhang22** [comment]
@VerisimilitudeX can you look into why there are now over 2000 changes in this PR?

**Comment by VerisimilitudeX** [comment]
Uhh, I think that happened when I tried syncing the pr to the master branch… I’m not quite sure how to fix it. Should I delete this pr and make a new one that is up to date with master? Also, Bea wanted me to rename master to main – when do you want me to do that?

Thanks,
Piyush

From: Oliver Zhang ***@***.***>
Date: Thursday, September 14, 2023 at 12:36 PM
To: NuevoFoundation/workshops ***@***.***>
Cc: Piyush Acharya ***@***.***>, Mention ***@***.***>
Subject: Re: [NuevoFoundation/workshops] GitHub basics (PR #393)

@VerisimilitudeX<https://github.com/VerisimilitudeX> can you look into why there are now over 2000 changes in this PR?

**Comment by VerisimilitudeX** [comment]
@ozhang22 I have fixed the 3000 changes issue and will start working on the story/adventure aspect tomorrow.

**Comment by VerisimilitudeX** [comment]
Do you know why the NuevoFoundation.workshops.PR might be failing?
image

**Comment by VerisimilitudeX** [comment]
Closing this PR and opening another one soon (fixes persistent merge conflicts).

**Pull request closed without merging**